### ISSUE TITLE
[Bug]: Report error in execution for scalar subqueries with more than 1 row

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#3303]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.5.0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

create table t1(a int, b int, c int);
insert into t1 values(null,null,null),(2,3,4),(4,5,6);

WITH qn AS (SELECT b as a FROM t1)
SELECT (WITH qn2 AS (SELECT a FROM qn WHERE a IS NULL or a>0)
        SELECT qn2.a FROM qn2) FROM qn;

[ERROR]
[SCRIPT   FILE]: cases/expression/cte.sql
[ROW    NUMBER]: 150
[SQL STATEMENT]: WITH qn AS (SELECT b as a FROM t1)
SELECT (WITH qn2 AS (SELECT a FROM qn WHERE a IS NULL or a>0)
SELECT qn2.a FROM qn2) FROM qn;
[EXPECT RESULT]:
Subquery returns more than 1 row
[ACTUAL RESULT]:
(with qn2 as (select a from qn where a is null or a > 0) select qn2.a from qn2)
null
3
5
null
3
5
null
3
5

**Comment by aunjgr** [comment]
Should implement SINGLE join to handle this case.

**Comment by domingozhang** [comment]
this will be fixed in 0.6

**Comment by aressu1985** [comment]
mysql> create table t1(a int, b int, c int);
Query OK, 0 rows affected (0.00 sec)

mysql> insert into t1 values(null,null,null),(2,3,4),(4,5,6);
Query OK, 3 rows affected (0.01 sec)

mysql> 
mysql> WITH qn AS (SELECT b as a FROM t1)
    -> SELECT (WITH qn2 AS (SELECT a FROM qn WHERE a IS NULL or a>0)
    -> SELECT qn2.a FROM qn2) FROM qn;
ERROR 20101 (HY000): internal error: scalar subquery returns more than 1 row
